feat(pii): Adopt new selectors

flub opened the pull request:
This adds new selectors for the minidump for the heap and stack.

@untitaker as the new tests show this sadly means any generic rule for `$binary` will apply to the stack.  Which is what we were trying to avoid.  So I'm not sure this proposed change will work.  I guess the question is whether we still want to this for just the heap or not.

Commit:
feat(pii): Adopt new selectors

This adds new selectors for the minidump for the heap and stack.

untitaker commented on relay-general/src/pii/minidumps.rs:
```diff
@@ -256,39 +256,61 @@ impl PiiAttachmentsProcessor<'_> {
                     let slice = data
                         .get_mut(range)
                         .ok_or(ScrubMinidumpError::InvalidAddress)?;
+                    let attrs = Cow::Owned(FieldAttrs::new().pii(Pii::Maybe));
```
You don't have to reinstantiate this I believe? Shouldn't you be able to pass down a Cow::Borrowed for the individual visits?

flub replied:
I can't pass this to multiple calls and it also is not clonable.  And since in the future we'll want some to be set to Pii::Yes de-duplicating this seems more cognitive overhead than it saves.  That was the logic anyway, I don't feel strongly so could change this.

untitaker replied:
FieldAttrs is Copy

untitaker commented on relay-general/src/pii/minidumps.rs:
```diff
@@ -542,4 +592,133 @@ mod tests {
+            serde_json::json!(
+                {
+                    "applications": {
+                        "$minidump:*": ["@anything:mask"],
```
```suggestion
                        "$minidump.*": ["@anything:mask"],
```

untitaker commented on relay-general/src/pii/minidumps.rs:
```diff
@@ -542,4 +592,133 @@ mod tests {
+            serde_json::json!(
+                {
+                    "applications": {
+                        "$attachments:**": ["@anything:mask"],
```
```suggestion
                        "$attachments.**": ["@anything:mask"],
```

untitaker commented on relay-general/src/pii/minidumps.rs:
```diff
@@ -542,4 +592,133 @@ mod tests {
+            ),
+        );
+        for stack in scrubber.stacks(Which::Scrubbed) {
+            assert!(!stack.iter().all(|b| *b == '*' as u8));
```
assert not (`!`) all, does that assert that the stack got partially scrubbed or not at all?

same for other assertions

untitaker reviewed (changes requested):
preliminary review... Pii::Maybe should prevent `$binary` from applying, not sure why it wouldn't work... but the tests seem broken

Commit:
Correct selectors and assertions

Commit:
Use byte literals

Thanks clippy.

untitaker committed:
fix pii=maybe support in attachment scrubbing

untitaker committed:
remove outdated comment

Commit:
Some tests for which selectors are specific

untitaker committed:
change maybe to true

untitaker commented on relay-general/src/pii/minidumps.rs:
```diff
@@ -542,4 +592,149 @@ mod tests {
+            serde_json::json!(
+                {
+                    "applications": {
+                        "$minidump.$stack_memory": ["@anything:mask"],
```
This selector doesn't have to work, customers currently use `$stack_memory`

Commit:
Make tests pass

With some justifications.  We'll just have to document these things as
known issues.

Commit:
Add changelog

Commit:
Merge branch 'master' into feat/minidump-selectors
